tests/accessor_legacy/accessor_api_utility.h

```diff
@@ -731,12 +732,12 @@ T get_zero_dim_buffer_value() {
  *        If (dims == 0), the data will be initialized using the common value
  *        for testing a zero-dim accessor.
  *        If false, data will be zero initialized.
- * @return Initialized data container
+ * @return unique_ptr to initialized array
  */
 template <typename T>
-std::vector<T> get_buffer_input_data(size_t count, int dims,
-                                                bool useIndexes = true) {
-  auto data = std::vector<T>(count);
+std::unique_ptr<T[]> get_buffer_input_data(size_t count, int dims,
+                                           bool useIndexes = true) {
+  auto data = std::make_unique<T[]>(count);
```

Comment:
Some alternative design could have been:
```c++
auto get_buffer_input_data(size_t count, int dims,
                                                bool useIndexes = true) {
  using return_type = std::vector<std::conditional_t<std::is_same_v<T, bool>, std::int8_t, T>>;
  auto data = return_type(count);
```
and then I have the feeling that there is no need to change anything else in the code.

Or even restructure the code for not using `get_buffer_input_data` at all but create an uninitialized buffer and initialize the data with an `host_accessor` on it, check the results with an `host_accessor` on it.
So no more `std::vector` at all. :-)
But this is probably too much for this PR.

Reply:
I think alternative design will make the code more complex because we have to cast raw pointer of vector from `int8_t` to `bool` , so there are two possible implementations: add type checks or use `reinterpret_cast`  on each buffer constructor, whereas I use just `data.get()`.

Restructuring will make the code even more complex. `get_buffer_input_data` is used many times, so we will replace this function with another using `host_accessor` and change every buffer initialization.

Reply:
OK. Thanks for the argumentation.